src/test/java/org/folio/rest/impl/EholdingsProvidersImplTest.java

```diff
@@ -0,0 +1,139 @@
+
+    spec = new RequestSpecBuilder()
+      .setBaseUri(host + ":" + port)
+      .build();
+  }
+
+  @Before
+  public void setUp() {
+    RMAPIConfigurationCache.getInstance().invalidate();
+  }
+
+  @Test
+  public void shouldReturnProvidersOnGet() throws IOException, URISyntaxException {
+    String stubResponseFile = "responses/rmapi/vendors/get-vendors-response.json";
+    int expectedTotalResults = 115;
+    String id = "131872";
+    String name = "Editions de L'Université de Bruxelles";
+    int packagesTotal = 1;
+    int packagesSelected = 0;
+    boolean supportsCustomPackages = false;
+
+    String wiremockUrl = host + ":" + userMockServer.port();
+    TestUtil.mockConfiguration("responses/configuration/get-configuration.json", wiremockUrl);
+    WireMock.stubFor(
+      WireMock.get(new UrlPathPattern(new RegexPattern("/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/vendors.*"), true))
+        .willReturn(new ResponseDefinitionBuilder()
+          .withBody(TestUtil.readFile(stubResponseFile))));
+    RestAssured.given()
+      .spec(spec).port(port)
+      .header(new Header(RestConstants.OKAPI_URL_HEADER, wiremockUrl))
+      .header(TENANT_HEADER)
+      .header(TOKEN_HEADER)
+      .when()
+      .get("eholdings/providers?q=e&page=1&sort=name")
+      .then()
+      .statusCode(200)
+      .body("meta.totalResults" , equalTo(expectedTotalResults))
+      .body("data[0].type", equalTo("providers"))
+      .body("data[0].id", equalTo(id))
+      .body("data[0].attributes.name", equalTo(name))
+      .body("data[0].attributes.packagesTotal", equalTo(packagesTotal))
+      .body("data[0].attributes.packagesSelected", equalTo(packagesSelected))
+      .body("data[0].attributes.supportsCustomPackages", equalTo(supportsCustomPackages));
+  }
+
+  @Test
+  public void shouldReturnErrorIfParameterInvalid() {
+    RestAssured.given()
+      .spec(spec).port(port)
+      .header(new Header(RestConstants.OKAPI_URL_HEADER, "http://localhost:8080"))
+      .header(TENANT_HEADER)
+      .header(TOKEN_HEADER)
+      .when()
+      .get("eholdings/providers?q=e&count=1000")
+      .then()
+      .statusCode(400)
+      .body("errors.first.title" , notNullValue());
+  }
+
+  @Test
+  public void shouldReturn500IfRMApiReturnsError() throws IOException, URISyntaxException {
+    String wiremockUrl = host + ":" + userMockServer.port();
+    TestUtil.mockConfiguration("responses/configuration/get-configuration.json", wiremockUrl);
+    WireMock.stubFor(
+      WireMock.get(new UrlPathPattern(new RegexPattern("/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/vendors.*"), true))
+        .willReturn(new ResponseDefinitionBuilder()
+          .withStatus(500)));
+
+    RestAssured.given()
+      .spec(spec).port(port)
+      .header(new Header(RestConstants.OKAPI_URL_HEADER, wiremockUrl))
+      .header(TENANT_HEADER)
+      .header(TOKEN_HEADER)
+      .when()
+      .get("eholdings/providers?q=e&count=1")
+      .then()
+      .statusCode(500)
```

Comment:
As mentioned in a comment above, not sure if we should treat all RM API error responses as 500's in our code since some of them might be informative for the end-user?